### PR TITLE
Deactivate

### PR DESCRIPTION
**Pull request by trapacska**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -275,23 +285,37 @@ func readLocalisedRecentChanges(recentChangesDir string) (map[string]string, err
 	return recentChangesMap, nil
 }
 
+func heighestVersionCode(versions []int64) (int64, error) {
```
Comment (godrei) [inline main.go]: i think we do not use this method

File: main.go [hunk 2]
```diff
@@ -422,39 +436,155 @@ func main() {
...
+		untrackApks = false
+	}
+
+	anyTrackUpdated := false
+
+	if untrackApks {
+		fmt.Println()
+		log.Infof("Deactivating blocking apk versions")
+
+		// Determin heighest version code
```
Comment (godrei) [inline main.go]: please remove the comments

File: main.go [hunk 3]
```diff
@@ -422,39 +436,155 @@ func main() {
...
+				} else {
+					track.VersionCodes = []int64{}
+					track.NullFields = []string{"VersionCodes"}
+				}
+
+				track.ForceSendFields = []string{"VersionCodes"}
+
+				tracksUpdateCall := tracksService.Patch(configs.PackageName, appEdit.Id, trackName, track)
+				if track, err := tracksUpdateCall.Do(); err != nil {
+					log.Printf("headers: %#v", tracksUpdateCall.Header())
```
Comment (godrei) [inline main.go]: please do not print this

File: main.go [hunk 4]
```diff
@@ -422,39 +436,155 @@ func main() {
...
+					log.Printf("track: %#v", track)
```
Comment (godrei) [inline main.go]: please do not print this

File: main.go [hunk 5]
```diff
@@ -422,39 +424,145 @@ func main() {
...
+		listResponse, err := tracksListCall.Do()
+		if err != nil {
+			failf("Failed to list tracks, error: %s", err)
+		}
+
+		tracks := listResponse.Tracks
+
+		possibleTrackNamesToUpdate := []string{}
+		switch configs.Track {
+		case "beta":
```
Comment (godrei) [inline main.go]: please use the defined constants e.g.: `alphaTrackName`, `betaTrackName`, ...

File: main.go [hunk 6]
```diff
@@ -422,39 +424,145 @@ func main() {
...
+		log.Printf(" possible tracks to update: %v", trackNamesToUpdate)
+
+		for _, trackName := range trackNamesToUpdate {
+			tracksGetCall := tracksService.Get(configs.PackageName, appEdit.Id, trackName)
+			track, err := tracksGetCall.Do()
+			if err != nil {
+				failf("Failed to get track (%s), error: %s", trackName, err)
+			}
+
+			log.Printf(" checking apk version on track: %s", track.Track)
```
Comment (godrei) [inline main.go]: version  -> versions

File: main.go [hunk 7]
```diff
@@ -422,39 +424,145 @@ func main() {
...
+				} else {
+					track.VersionCodes = []int64{}
+					track.NullFields = []string{"VersionCodes"}
+				}
+
+				track.ForceSendFields = []string{"VersionCodes"}
+
+				tracksUpdateCall := tracksService.Patch(configs.PackageName, appEdit.Id, trackName, track)
+				if track, err := tracksUpdateCall.Do(); err != nil {
+					if err == io.EOF {
```
Comment (godrei) [inline main.go]: do not print the eof error, it is not usefull i think

```
if err != nil && err != ios.EOF {
  failf("Failed to update track (%s), error: %s", trackName, err)
}
```

Comment (trapacska) [inline main.go]: Nice!

**Comment by godrei** [comment]
🚀

**Pull request merged**